### PR TITLE
Add font from theme to the root of the TextField.

### LINKED ISSUE #8007: TextField: fonts not respecting theme font.
https://codepen.io/dzearing/pen/ZwvyNq

image

The top TextField is rendered within a `Fabric` component, defining a default `font-family`. The bottom one is without.

Expected:
They should match.

Resulted:
They don't.

### PR DESCRIPTION
**Pull request by aneeshack4**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8007 
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Added theme.font.medium to the root of Textfield's styles so that it inherits font from theme. Also added Fabric wrapper to stories to avoid changing these screener scenarios.

#### Focus areas to test

Screener scenarios for Textfield, focusing on font.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8018)

### CODE REVIEW

File: common/changes/office-ui-fabric-react/textfield_font_2019-02-15-23-14.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Loading font from theme for Textfield.",
```
Comment (Vitalius1) [inline common/changes/office-ui-fabric-react/textfield_font_2019-02-15-23-14.json]:
```suggestion
      "comment": "TextField: adds font styles passed from theme to the root element.",
```

**Comment by Vitalius1** [comment]
@aneeshack4 can you elaborate on this:

> Also added Fabric wrapper to stories to avoid changing these screener scenarios.

I don't get it. Do we not want the screener to reflect the current state of the styles?

File: common/changes/office-ui-fabric-react/textfield_font_2019-02-15-23-14.json (hunk 1 above)
Comment (aneeshack4) [reply common/changes/office-ui-fabric-react/textfield_font_2019-02-15-23-14.json]: I'll commit this suggestion once I finish fixing the build failures.

**Comment by aneeshack4** [comment]
> @aneeshack4 can you elaborate on this:
> 
> > Also added Fabric wrapper to stories to avoid changing these screener scenarios.
> 
> I don't get it. Do we not want the screener to reflect the current state of the styles?



> @aneeshack4 can you elaborate on this:
> 
> > Also added Fabric wrapper to stories to avoid changing these screener scenarios.
> 
> I don't get it. Do we not want the screener to reflect the current state of the styles?

So in the past, when I did the same thing for all other Fabric components in https://github.com/OfficeDev/office-ui-fabric-react/pull/7276, a bunch of the Screener tests failed where font wasn't defined in the stories (just like with Textfield's stories file) and I had to add Fabric wrappers to make them pass. They're passing when I ran "npm run screener:local" with this Fabric wrapper.

**Comment by aneeshack4** [comment]
Ah actually with the updated commits, the local screener results show some changes in Textfield. I believe the font-family & font-size updates are expected so those I think will need to be looked at for approval (just as we did in #7276). I'm looking into the color change. There's one small Callout change that I'll fix as well.

**Comment by Vitalius1** [comment]
Good to know this... It just feels weird when the title of the #7276 sounds like you are removing the use of the `Fabric` component but then in stories we actually need them. Do we know what is the actual issue why stories are behaving differently? Would be nice to have this documented in the Tactical OneNote so other developers know about this issue when adding new storries or theme.font to component styles.

**Comment by dzearing** [comment]
I went through the screener tests and see a number of issues:

1. placeholder text is still rendering as arial in places.
2. placeholder text seems to have disappeared in some cases.
3. There is a "disabled" test that seems to have changed the color of the label.

Take a look at the screener changes. No colors should have been changed, just font families and sizes.

@Vitalius1 The only case we should be using the Fabric component are scss cases where they don't use contextual theming.

**Comment by dzearing** [comment]
@aneeshack4 I know you're pretty busy with shield, but please give me a heads up when this is fixed.

**Comment by aneeshack4** [comment]
@dzearing Yes apologies for the delay - it's been pretty crazy with all the incoming issues on Shield. I did try to fix the screener disabled scenarios by adding 'theme.fonts.medium' to the disabled styles but when I tested it locally it broke more things which is why I haven't pushed my local changes - I will try my best to get back to this today.

**Comment by aneeshack4** [comment]
Took another stab at this - there are 12 rejected changes and all of them have to do with the prefix:
image
So I added theme.fonts.medium to the prefix styling in TextField.styles.tsx and it didn't change anything so I'm not really sure what to do now.

Also @dzearing, I'm confused now re-reading the 2nd of the 3 issues above - which ones are you seeing that now have Arial? All of the 12 rejected failures now have Segoe which is correct, right?

And I'm at a complete loss for why the text has completely disappeared in some cases - I believe this is is what you're referring to: 
image

As for the label's color, there is no "label: []" like there is for field, prefix, suffix, etc... so I'm really confused as to why adding the one theme.fonts.medium at root would cause this and how to go about overriding it.

**Comment by dzearing** [comment]
Resetting PR to see if there was some sort of merge anomaly causing screener regressions.

**Comment by aneeshack4** [comment]
Merged with latest master and the local build succeeded, pushed to see if that changes screener results but instead the build failed... closing and reopening to trigger again.

**Comment by aneeshack4** [comment]
Looks like most of the screener changes now are expected except for the value of the textfield in some of the prefix scenarios which disappeared - looking into why

**Pull request merged**